engine/src/main/java/io/zeebe/engine/processor/workflow/message/BufferedMessageToStartEventCorrelator.java

```diff
@@ -83,10 +85,12 @@ private Correlation findNextMessageToCorrelate(
     for (final ExecutableStartEvent startEvent : workflow.getWorkflow().getStartEvents()) {
       if (startEvent.isMessage()) {
 
-        final var messageName = startEvent.getMessage().getMessageName();
+        final Optional<String> optMessageName = startEvent.getMessage().getMessageName();
+        final org.agrona.DirectBuffer messageNameBuffer = new UnsafeBuffer();
+        messageNameBuffer.wrap(optMessageName.get().getBytes());
```

Comment:
Using `BufferUtil.wrapString()`:

```suggestion
        final DirectBuffer messageNameBuffer = startEvent
            .getMessage()
            .getMessageName()
            .map(BufferUtil::wrapString)
            .orElseThrow();
```